src/content/docs/it/guides/backend/google-firebase.mdx

```diff
@@ -0,0 +1,915 @@
+
+```ts title="src/pages/api/auth/signin.ts"
+import type { APIRoute } from "astro";
+import { app } from "../../../firebase/server";
+import { getAuth } from "firebase-admin/auth";
+
+export const get: APIRoute = async ({ request, cookies, redirect }) => {
+  const auth = getAuth(app);
+
+  /* Ottieni il token dagli headers della richiesta */
+  const idToken = request.headers.get("Authorization")?.split("Bearer ")[1];
+  if (!idToken) {
+    return new Response(
+      "Nessun token trovato",
+      { status: 401 }
+    );
+  }
+
+  /* Verifica il token */
+  try {
+    await auth.verifyIdToken(idToken);
+  } catch (error) {
+    return new Response(
+      "Token non valido",
+      { status: 401 }
+    );
+  }
+
+  /* Crea e imposta il cookie per la sessione */
+  const fiveDays = 60 * 60 * 24 * 5 * 1000;
+  const sessionCookie = await auth.createSessionCookie(idToken, {
+    expiresIn: fiveDays,
+  });
+
+  cookies.set("session", sessionCookie, {
+    path: "/",
+  });
+
+  return redirect("/dashboard");
+};
+```
+
+:::note
+Questa è un'implementazione di base dell'endpoint per l'accesso. Puoi aggiungere ulteriore logica a questo endpoint per adattarlo alle tue esigenze.
+:::
+
+`signout.ts` contiene il codice per disconnettere un utente eliminando il cookie per la sessione:
+
+```ts title="src/pages/api/auth/signout.ts"
+import type { APIRoute } from "astro";
+
+export const get: APIRoute = async ({ redirect, cookies }) => {
+  cookies.delete("session", {
+    path: "/",
+  });
+  return redirect("/signin");
+};
+```
+
+:::note
+Questa è un'implementazione di base del endpoint per la disconnessione. Puoi aggiungere ulteriore logica a questo endpoint per adattarlo alle tue esigenze.
+:::
+
+`register.ts` contiene il codice per registrare un utente utilizzando Firebase:
+
+```ts title="src/pages/api/auth/register.ts"
+import type { APIRoute } from "astro";
+import { getAuth } from "firebase-admin/auth";
+import { app } from "../../../firebase/server";
+
+export const post: APIRoute = async ({ request, redirect }) => {
+  const auth = getAuth(app);
+
+  /* Ottieni i dati del form */
+  const formData = await request.formData();
+  const email = formData.get("email")?.toString();
+  const password = formData.get("password")?.toString();
+  const name = formData.get("name")?.toString();
+
+  if (!email || !password || !name) {
+    return new Response(
+      "Alcuni dati del form mancano",
+      { status: 400 }
+    );
+  }
+
+  /* Crea l'utente */
+  try {
+    await auth.createUser({
+      email,
+      password,
+      displayName: name,
+    });
+  } catch (error: any) {
+    return new Response(
+      "Qualcosa è andato storto",
+      { status: 400 }
+    );
+  }
+  return redirect("/signin");
+};
+```
+
+:::note
+Questa è un'implementazione di base del endpoint per la registrazione. Puoi aggiungere ulteriore logica a questo endpoint per adattarlo alle tue esigenze.
```

Comment:
I found multiple occurrences of `del endopoint`... Shouldn't it be `dell'endpoint` instead? Same for `al endpoint` -> `all'endpoint`?
E.g. at line 554 there's an example of what I think it's the correct form.